yarn-project/aztec-node/src/aztec-node/http-node.ts

```diff
@@ -370,4 +371,19 @@ export class HttpNode implements AztecNode {
     const response = await (await fetch(url.toString())).json();
     return response.blockData;
   }
+
+  /**
+   * Simulates the public part of a transaction with the current state.
+   * @param tx - The transaction to simulate.
+   **/
+  public async simulatePublicPart(tx: Tx) {
```

Comment:
Aside from the name, does it make sense to send a full tx object? Shouldn't we be just sending the `enqueuedFunctionCall`s, plus any side effects from private execution that may affect the public simulation (such as new contract deployment)?

Reply:
I think it's better to send the whole TX for some reasons:
 - This method IMO will evolve to be the gas estimator of a TX. In ethereum, gas estimation takes TX objects https://www.quicknode.com/docs/ethereum/eth_estimateGas
 - This also runs the public kernel simulator, which I think makes this method need most of the fields in the TX
 - It just feels like a cleaner interface to me. Create your TX, call the node to see if it will succeed / to estimate fees on it!

Reply:
Fair point. And I know this is probably a bigger discussion, but how does this impact privacy? Should we enable the RPC server to run its own public simulations and gas estimations, by just requiring the state it needs from the node? Or would that leak information as well?

Reply:
I believe it's ok, since you send to the node for simulation the same data you'd send the node for submission of the TX. I think in this case it's similar to ethereum, if you don't want the node to track which transaction came from which IP, you can run the node yourself and rely on the gossip p2p networking.